query-node/schemas/bounty.graphql

```diff
@@ -0,0 +1,190 @@
+"Any member can submit work entries to the bounty"
+type BountyContractOpen @variant {
+  # no properties
+
+  # TODO: remove me - variant needs to have at least 1 property now
+  dummy: Int
+}
+
+"Only preselected members can submit work entries to the bounty"
+type BountyContractClosed @variant {
+  # Members authorized to submit entries to the bounty
+  whitelist: [Membership!]!
+}
+
+"Valid of type assurance contract"
+union BountyContractType = BountyContractOpen | BountyContractClosed
+
+type BountyFundingPerpetual @variant {
+  "Desired funding"
+  target: BigInt!
+}
+
+type BountyFundingLimited @variant {
+  "Minimum amount of funds for a successful bounty"
+  minFundingAmount: BigInt!
+
+  "Upper boundary for a bounty funding"
+  maxFundingAmount: BigInt!
+
+  "Maximum allowed funding period"
+  fundingPeriod: Int!
+}
+
+"Bounty funding types"
+union BountyFundingType = BountyFundingPerpetual | BountyFundingLimited
+
+"All valid bounty stages"
+enum BountyStage {
+  Funding
+  Expired
+  WorkSubmission
+  Judgment
+  Successful
+  Failed
+  Terminated
+}
+
+type Bounty @entity {
+  # Bounty creation parameters:
+  "Bounty's runtime id"
+  id: ID!
+
+  "Bounty title"
+  title: String!
+
+  "Bounty description"
+  description: String!
+
+  "Amount of funding provided by the creator"
+  cherry: BigInt!
+
+  "Stake minimum amount required to submit work entry to the bounty"
+  entrantStake: BigInt!
+
+  "Bounty creator"
+  creator: Membership
+
+  "Bounty oracle"
+  oracle: Membership
+
+  "Bounty funding type"
+  fundingType: BountyFundingType!
+
+  "Bounty assurance contract type"
+  contractType: BountyContractType!
+
+  "Number of blocks from end of funding period until people can no longer submit bounty submissions"
+  workPeriod: Int!
+
+  "Number of block from end of work period until oracle can no longer decide winners"
+  judgingPeriod: Int!
+
+  # Bounty current state:
+  "Current bounty stage"
+  stage: BountyStage!
+
+  totalFunding: BigInt!
+
+  "Bounty's discussion thread"
+  discussionThread: ForumThread!
```

Comment:
According to the design there is a discussion thread under the "Bounty tab". But the extrinsics don't refer to this. Should a forum thread be created before calling `createBounty` and the threadId be stored in the metadata ?

Reply:
The extrinsic will not have this for sure. The metadata should have this.

AFAIR there should be a predefined forum category in which we would create a thread. How to get this is another story for sure ;)

Reply:
BTW: #2961